src/api/PDFDocument.ts

```diff
@@ -713,6 +713,29 @@ export default class PDFDocument {
     return copiedPages;
   }
 
+  /**
+   * Add document JavaScript. The script is executed when the document is opened.
+   * See the [JavaScript™ for Acrobat® API Reference](https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/js_api_reference.pdf)
+   * for details.
+   * @param script The JavaScript to execute.
+   * @param name The name of the script. Must be unique per document.
+   */
+  addJavascript(script: string, name: string) {
```

Comment:
Can we alter the implementation of this method a bit? There are some existing patterns in the codebase that should be followed. Please see https://github.com/Hopding/pdf-lib/blob/master/src/api/PDFEmbeddedFile.ts#L48. File embedding is similar to JavaScript embedding in some respects.

Reply:
I've taken the code you suggested and adopted it. It's way better this way...

Reply:
Thanks! I'll try to review it again tomorrow evening.